File: Scripts/server/inheritance-single/perl/inheritance-single.pl
use MooseX::Declare;

class Animal {
    # methods go here...
    sub speak {
        print "Animal speaking";
    }

    sub walk {
        print "Animal walking\n";
    }
}
class Dog extends Animal {
    # methods go here...
    sub speak {
        print "Dog barking\n";
    }
}
class Cat extends Animal {
    # methods go here...
        sub speak {
        print "Cat meawing";
    }
}
class Lab extends Dog {
    # methods go here...
        sub speak {
        print "Lab barking";
    }
}
class Collie extends Dog {
    # methods go here...
        sub speak {
        print "Collie barking";
    }
}

for( $a=0; $a < 1000000; $a = $a + 1) {
    my $obj = Dog->new();
    $obj->speak();
    $obj->walk();
}

File: Scripts/server/inheritance-single/python/inheritance-single.py
class Animal:
  
    def speak(self):
      print "Animal speaking"

    def walk(self):
      print "Animal walking"

class Dog(Animal):

    def speak(self):
      print "Dog barking"
      self.walk()

class Cat(Animal):

  def speak():
      print "Cat meawing"

class Lab(Dog):

  def speak():
      print "Lab barking"

class Collie(Dog):

  def speak():
      print "Collie barking"

for i in range(1000000):
	obj = Dog()
	obj.speak()
  #obj.walk()
